src/documents/bulk_download.py

```diff
@@ -37,32 +44,38 @@ def _formatted_filepath(
         doc: Document,
         archive: bool = False,
         folder: str = "",
-    ):
+    ) -> Path:
         """
         Constructs a full file path for the given document to be used inside
         the zipfile.
 
         The path is already unique, as handled when a document is consumed or updated
         """
         if archive and doc.has_archive_version:
-            in_archive_path = os.path.join(folder, doc.archive_filename)
+            if doc.archive_filename is None:
+                raise ValueError("doc %a has no archive_filename", doc)
+            in_archive_path: Path = Path(folder) / doc.archive_filename
         else:
-            in_archive_path = os.path.join(folder, doc.filename)
+            if doc.filename is None:
```

Comment:
I don't think the `filename` of a model can ever be None either, even though the DB column is nullable, it is always assigned to something or we'd lose the file.